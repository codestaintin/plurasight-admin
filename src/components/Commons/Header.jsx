import React from 'react';
import { shape, arrayOf, bool } from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

/**
 *
 * @function Header
 *
 * @returns {JSX}
 */
const Header = ({ courses }) => (
  <nav className="navbar navbar-expand-lg navbar-dark bg-dark custom_nav">
    <ul className="navbar-nav">
      <li className="nav-item dropdown">
        <Link className="nav-link" to="/">Home</Link>
      </li>
      <li className="nav-item">
        <Link className="nav-link" to="/about">About</Link>
      </li>
      <li className="nav-item">
        <Link className="nav-link" to="/courses">
          Courses <span className="badge badge-light">{courses.length}</span>
        </Link>
      </li>
    </ul>
  </nav>
);

Header.propTypes = {
  courses: arrayOf(shape({}))
};

const mapStateToProps = ({ allCourses }) => ({
  courses: allCourses.courses
});

export default connect(mapStateToProps)(Header);
