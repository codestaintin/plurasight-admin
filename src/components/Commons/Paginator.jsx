import React from 'react';
import Paginate from 'react-paginate';
import { number, func } from 'prop-types';

const Paginator = ({ pageCount, page, handlePageClick }) => (
  <Paginate
    pageCount={pageCount}
    pageRangeDisplayed={2}
    initialPage={page - 1}
    marginPagesDisplayed={2}
    disableInitialCallback
    onPageChange={handlePageClick}
    containerClassName="pagination justify-content-center"
    subContainerClassName="page-item"
    pageLinkClassName="page-link"
    activeClassName="page-item active"
    nextLinkClassName="page-link"
    previousLinkClassName="page-link"
  />
);

Paginator.propTypes = {
  page: number,
  pageCount: number,
  handlePageClick: func.isRequired
};
export default Paginator;
