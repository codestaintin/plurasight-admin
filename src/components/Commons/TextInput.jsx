import React from 'react';
import PropTypes from 'prop-types';

const TextInput = ({
  name, label, onChange, placeholder, value, error, pattern
}) => {
  let wrapperClass = 'form-group';
  if (error && error.length > 0) {
    /*eslint-disable */
    wrapperClass += ' ' + 'has-error';
  }

  return (
    <div className={wrapperClass}>
      <label htmlFor={name}>{label}</label>
      <div className="field">
        <input
          type="text"
          name={name}
          className="form-control"
          placeholder={placeholder}
          pattern={pattern}
          value={value}
          onChange={onChange} />
        {error && <div className="alert alert-danger">{error}</div>}
      </div>
    </div>
  );
};

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  pattern: PropTypes.string,
  value: PropTypes.string,
  error: PropTypes.string
};

export default TextInput;
