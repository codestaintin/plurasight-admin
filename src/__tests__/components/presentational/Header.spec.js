import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import { Header } from '../../../components/Commons/Header';

configure({ adapter: new Adapter() });

describe('<Header>', () => {
  const props = {
    courses: [{
      id: 'react-flux-building-applications',
      title: 'Building Applications in React and Flux',
      watchHref: 'http://www.pluralsight.com/courses/react-flux',
      authorId: 'cory-house',
      length: '5:08',
      category: 'JavaScript'
    }]
  };

  it('renders the <Header> component', () => {
    const tree = shallow(<Header {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
