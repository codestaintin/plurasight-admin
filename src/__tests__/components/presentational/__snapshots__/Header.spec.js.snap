// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> renders the <Header> component 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark custom_nav"
>
  <ul
    className="navbar-nav"
  >
    <li
      className="nav-item dropdown"
    >
      <Link
        className="nav-link"
        replace={false}
        to="/"
      >
        Home
      </Link>
    </li>
    <li
      className="nav-item"
    >
      <Link
        className="nav-link"
        replace={false}
        to="/about"
      >
        About
      </Link>
    </li>
    <li
      className="nav-item"
    >
      <Link
        className="nav-link"
        replace={false}
        to="/courses"
      >
        Courses 
        <span
          className="badge badge-light"
        >
          1
        </span>
      </Link>
    </li>
  </ul>
</nav>
`;
